openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedCardV2/ActivityFeedCardV2.tsx

```diff
@@ -63,31 +77,130 @@ const ActivityFeedCardV2 = ({
   const onUpdate = (message: string) => {
     const updatedPost = { ...feed, message };
     const patch = compare(feed, updatedPost);
-    updateFeed(feed.id, post.id, !isPost, patch);
+    if (isAnnouncementTab) {
+      // update post in announcement tab
+      updatePost(feed.id, post.id, patch);
+      updateAnnouncementThreads && updateAnnouncementThreads();
+    } else {
+      updateFeed(feed.id, post.id, !isPost, patch);
+    }
+
     setIsEditPost(!isEditPost);
   };
 
+  const updateThreadHandler = async (
+    threadId: string,
+    postId: string,
+    isThread: boolean,
+    data: Operation[]
+  ): Promise<void> => {
+    const callback = () => {
+      return;
+    };
+    await updateThreadData(threadId, postId, isThread, data, callback);
+    await fetchUpdatedThread(threadId);
+  };
+  const handleAnnouncementUpdate = async (
+    title: string,
+    announcement: AnnouncementDetails
+  ) => {
+    const existingAnnouncement = {
```

Comment:
This method will be called for announcement only no?
If so why we have checks here for isAnnouncement

Reply:
isAnnouncementTab is used to track the Announcement tab in Service page